internal/crd/cache_test.go

```diff
@@ -0,0 +1,76 @@
+package crd_test
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"github.com/kyma-project/lifecycle-manager/internal/crd"
+)
+
+const key = "testKey"
+
+func TestNewCache_WhenCalled_NoCachePassedIn(t *testing.T) {
+	cache := crd.NewCache(nil)
+
+	assert.NotNil(t, cache)
+}
+
+func TestNewCache_WhenCalled_CachePassedIn(t *testing.T) {
+	internalCache := &sync.Map{}
+	cache := crd.NewCache(internalCache)
+
+	assert.NotNil(t, cache)
+}
+
+func TestGet_WhenCalled_NotInCache(t *testing.T) {
+	cache := crd.NewCache(nil)
+
+	cachedCrd, ok := cache.Get(key)
+
+	assert.False(t, ok)
+	assert.Equal(t, apiextensionsv1.CustomResourceDefinition{}, cachedCrd)
+}
+
+func TestGet_WhenInCache_TypeIsWrong(t *testing.T) {
+	internalCache := &sync.Map{}
+	cache := crd.NewCache(internalCache)
+	internalCache.Store(key, "not a CRD")
+
+	cachedCrd, ok := cache.Get(key)
+
+	assert.False(t, ok)
+	assert.Equal(t, apiextensionsv1.CustomResourceDefinition{}, cachedCrd)
+}
+
+func TestGet_WhenInCache_TypeIsRight(t *testing.T) {
+	internalCache := &sync.Map{}
+	cache := crd.NewCache(internalCache)
+	someCrd := apiextensionsv1.CustomResourceDefinition{}
+	someCrd.Name = "some-crd"
+	internalCache.Store(key, someCrd)
+
+	cachedCrd, ok := cache.Get(key)
+
+	assert.True(t, ok)
+	assert.Equal(t, someCrd.Name, cachedCrd.Name)
+	assert.Equal(t, someCrd, cachedCrd)
```

Comment:
why comparing the names explicitly? Aren't structs "value types" where each value is compared individually, so it's covered by below check anyway?

Reply:
Ye the last one is enough! Forgot to delete the first assertion while trying things. Thanks, removed. 
Same for other comment.